src/use-auth.js

```diff
@@ -0,0 +1,86 @@
+import React, {
+  useState,
+  useContext,
+  createContext,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useQuery, useMutation, gql } from '@apollo/client';
+
+const REGISTER = gql`
+  mutation RegisterMutation(
+    $username: String!
+    $firstName: String!
+    $lastName: String!
+    $password: String!
+  ) {
+    createUser(
+      username: $username
+      firstName: $firstName
+      lastName: $lastName
+      password: $password
+      userRoleId: 3
+    ) {
+      id
+    }
+  }
+`;
+const QUERY_USERS = gql`
+{
+  users {
+    username
+    id
+    password
+  }
+}
+`;
+
+const authContext = createContext();
+
+const useAuth = () => useContext(authContext);
+
+function useProvideAuth() {
+  const [user, setUser] = useState(localStorage.getItem('user') || 0);
```

Comment:
`user` here doesn't suggest what it's for exactly and it's hard to read its structure as well, I would suggest to be more descriptive because it feels like an object, so maybe you could go with: `loggedUserId` or `currentUserId`,...